pkgs/development/compilers/flutter/patches/flutter310/move-cache.patch

```diff
@@ -0,0 +1,152 @@
+diff --git a/packages/flutter_tools/bin/podhelper.rb b/packages/flutter_tools/bin/podhelper.rb
+index 01d0bf4e21..cb00a44674 100644
+--- a/packages/flutter_tools/bin/podhelper.rb
++++ b/packages/flutter_tools/bin/podhelper.rb
+@@ -39,7 +39,7 @@ def flutter_additional_ios_build_settings(target)
+
+   # This podhelper script is at $FLUTTER_ROOT/packages/flutter_tools/bin.
+   # Add search paths from $FLUTTER_ROOT/bin/cache/artifacts/engine.
+-  artifacts_dir = File.join('..', '..', '..', '..', 'bin', 'cache', 'artifacts', 'engine')
++  artifacts_dir = File.join(ENV['HOME'], '.cache/flutter/artifacts/engine')
+   debug_framework_dir = File.expand_path(File.join(artifacts_dir, 'ios', 'Flutter.xcframework'), __FILE__)
+
+   unless Dir.exist?(debug_framework_dir)
+@@ -121,7 +121,7 @@ def flutter_additional_macos_build_settings(target)
+
+   # This podhelper script is at $FLUTTER_ROOT/packages/flutter_tools/bin.
+   # Add search paths from $FLUTTER_ROOT/bin/cache/artifacts/engine.
+-  artifacts_dir = File.join('..', '..', '..', '..', 'bin', 'cache', 'artifacts', 'engine')
++  artifacts_dir = File.join(ENV['HOME'], '.cache/flutter/artifacts/engine')
+   debug_framework_dir = File.expand_path(File.join(artifacts_dir, 'darwin-x64'), __FILE__)
+   release_framework_dir = File.expand_path(File.join(artifacts_dir, 'darwin-x64-release'), __FILE__)
+
+diff --git a/packages/flutter_tools/lib/src/artifacts.dart b/packages/flutter_tools/lib/src/artifacts.dart
+index a874b3c703..ba5ddd6600 100644
+--- a/packages/flutter_tools/lib/src/artifacts.dart
++++ b/packages/flutter_tools/lib/src/artifacts.dart
+@@ -643,7 +643,7 @@ class CachedArtifacts implements Artifacts {
+       case Artifact.dart2wasmSnapshot:
+       case Artifact.frontendServerSnapshotForEngineDartSdk:
+         return _fileSystem.path.join(
+-          _dartSdkPath(_cache), 'bin', 'snapshots',
++          _dartSdkPath(_fileSystem), 'bin', 'snapshots',
```

Comment:
why is it changed from _cache to _fileSystem in several places in the patch? And could the reason be documented somewhere? maybe in the git commit message